Gordon360/Controllers/HousingController.cs

```diff
@@ -229,4 +235,463 @@ public ActionResult<ApartmentApplicationViewModel[]> GetAllApartmentApplication(
             return NotFound();
         }
     }
+
+    /// <summary>
+    /// Creates a new hall assignment range if it does not overlap with any existing ranges
+    /// </summary>
+    /// <param name="model">The ViewModel that contains the hall ID and room range</param>
+    /// <returns>The created Hall_Assignment_Ranges object</returns>
+    [HttpPost("roomrange")]
+    [StateYourBusiness(operation = Operation.ADD, resource = Resource.HOUSING_ROOM_RANGE)]
+    public async Task<ActionResult<Hall_Assignment_Ranges>> CreateRoomRange([FromBody] HallAssignmentRangeViewModel model)
+    {
+        try
+        {
+            var newRange = await housingService.CreateRoomRangeAsync(model);
+            return Created("Room range created successfully.", newRange);
+        }
+        catch (InvalidOperationException ex)
+        {
+            return BadRequest(ex.Message);
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"Internal server error: {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Retrieves all room ranges.
+    /// </summary>
+    /// <returns>A list of room ranges.</returns>
+    [HttpGet("roomrange/all")]
+    [StateYourBusiness(operation = Operation.READ_ALL, resource = Resource.HOUSING_ROOM_RANGE)]
+    public async Task<ActionResult<List<HallAssignmentRangeViewModel>>> GetAllRoomRanges()
+    {
+        try
+        {
+            var roomRanges = await housingService.GetAllRoomRangesAsync();
+            return Ok(roomRanges);
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"Internal server error: {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Deletes a Room Range
+    /// </summary>
+    /// <param name="rangeId">The ID of the room range to delete</param>
+    /// <returns> Returns if completed</returns>
+    [HttpDelete("roomrange/{rangeId}")]
+    [StateYourBusiness(operation = Operation.DELETE, resource = Resource.HOUSING_ROOM_RANGE)]
+    public async Task<IActionResult> DeleteRoomRange(int rangeId)
+    {
+        try
+        {
+            var success = await housingService.DeleteRoomRangeAsync(rangeId);
+            if (success)
+            {
+                return Ok(new { message = "Room range deleted successfully." });
+            }
+            else
+            {
+                return BadRequest(new { message = "Failed to delete the room range." });
+            }
+        }
+        catch (ResourceNotFoundException ex)
+        {
+            return NotFound(new { message = ex.ExceptionMessage });
+        }
```

Comment:
Any uncaught `ResourceNotFoundException`s are automatically transformed into NotFound responses by our CustomExcetpions middleware, so this is unnecessary.

Reply:
removed unneeded try catches for resource not found exceptions in a separate branch